src/darwin/agent/reportcrash.js

```diff
@@ -96,45 +97,70 @@ Interceptor.attach(Module.getExportByName(CORESYMBOLICATION_PATH, 'task_is_64bit
     this.pid = pidForTask(args[0].toUInt32());
   },
   onLeave: function (retval) {
-    if (this.pid === crashedPid)
-      is64Bit = !!retval.toUInt32();
+    const localState = state[this.threadId];
+    if (localState === undefined) {
+      console.error('task_is_64bit: missing state for thread', this.threadId);
```

Comment:
Hmm, how about making a `getState(operation)`, where `operation` is e.g. `task_is_64bit`, that throws an exception formatted exactly like this when `undefined`. Unhandled JS exceptions are logged to `stderr`, so the result will be roughly the same, but will simplify the code a little bit.